clippy_lints/src/modulo_arithmetic.rs

```diff
@@ -0,0 +1,102 @@
+use crate::consts::{constant, Constant};
+use crate::utils::{sext, span_help_and_lint};
+use if_chain::if_chain;
+use rustc::declare_lint_pass;
+use rustc::hir::*;
+use rustc::lint::{LateContext, LateLintPass, LintArray, LintPass};
+use rustc::ty::{self};
+use rustc_session::declare_tool_lint;
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for modulo arithemtic.
+    ///
+    /// **Why is this bad?** The results of moudlo (%) operation migth differ
+    /// depending on the language, when negative numbers are involved.
+    /// If you interop with different languages it might be beneficial
+    /// to double check all places that use modulo arithmetic.
+    ///
+    /// For example, in Rust `17 % -3 = 2`, but in Python `17 % -3 = -1`.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    /// ```rust
+    /// let x = -17 % 3;
+    /// ```
+    pub MODULO_ARITHMETIC,
+    nursery,
+    "any modulo arithmetic statement"
+}
+
+declare_lint_pass!(ModuloArithmetic => [MODULO_ARITHMETIC]);
+
+fn is_negative<'a, 'tcx>(c: &ptr::P<rustc::hir::Expr>, cx: &LateContext<'a, 'tcx>, expr: &'tcx Expr) -> Option<bool> {
```

Comment:
```suggestion
fn is_negative(operand: &Expr, cx: &LateContext<'_, '_>, expr: &Expr) -> Option<bool> {
```

should work.

Reply:
Fixed in https://github.com/rust-lang/rust-clippy/pull/4867/commits/58d46176d766f96d9e6f00813e5cc704d4ad6476

That is actually a great comment.